### ISSUE TITLE
[Android] LocationServices disabled when missing one permission

### ISSUE THREAD

**Issue by mayardb** [mapbox/mapbox-gl-native#4083]
Hey MapBox Team!

I was trying to implement `Tracking` and `Bearing`, and I can't get any `Location`.
I have just find why: the LocationServices required both location permissions:

In `LocationServices` line 67:
```
public void toggleGPS(boolean enableGPS) {
      if ((ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) ||
                (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED)) {
            Log.w(TAG, "Location Permissions Not Granted Yet.  Try again after requesting.");
            return;
      }
      ...
}
```

I do not think that they are both required since ACCESS_FINE_LOCATION includes the ACCESS_COARSE_LOCATION as you can read here: http://developer.android.com/guide/topics/location/strategies.html
```
Note: If you are using both NETWORK_PROVIDER and GPS_PROVIDER, then you need to request only the ACCESS_FINE_LOCATION permission, because it includes permission for both providers. (Permission for ACCESS_COARSE_LOCATION includes permission only for NETWORK_PROVIDER.)
```

Is that possible that we change the `||` into `&&` to be able to use only one of them ?

Best regards.

**Comment by tobrun** [comment]
Thanks again for reaching out and writing up this issue @mayardb!
Makes a lot of sense relaxing that if check. Will pick this up.

**Comment by tobrun** [comment]
Going to cleanup LocationServices class:
 - rename `LocationServices` to `LocationService`
  - not seeing the requirement of making the name plural
  - the class itself is a singleton
 - currently multiple name conventions are being used in one class
 - cleanup some android studio warnings

cc @bleege

**Comment by bleege** [comment]
Couple of quick things here @tobrun ....

 - rename `LocationServices` to `LocationService`
  - not seeing the requirement of making the name plural
  - the class itself is a singleton
 - renamed `getLocationServices` to `getInstance` since it's a singleton pattern

While both are true and the instinct to name them this way is correct IMHO, this is one of those cases where I think it's more valuable to have things named the original way.  Here's why:

`LocationService` implies that that it's an Android `Service` of some type when it's not, it's a singleton as you note.  The plural implies that there are N number of Location related functions in it, which there are.  It also helps imply that this is centralized place for Location tech in the SDK which we want it to be.

As for `getLocationServices` vs `getInstance` again the Java tradition is to use `getInstance` but in this case it kind of runs against the "centralized place" concept I previously mentioned.  `getInstance` has a bit of connotation of a generic new version being created, while `getLocationServices` reinforces that this is more of a centralized spot for this tech in the SDK.

I know this is a bit semantic and nitpicky, but would you please restore these two to their original names?

**Comment by tobrun** [comment]
@bleege done in #4102, 
for next time I will PR such a change in a review first.

**Comment by bleege** [comment]
Rebased and merged #4102 as CI was happy.
